src/ResourceManager/EventGrid/Commands.EventGrid/Domain/GetAzureEventGridDomainTopic.cs

```diff
@@ -0,0 +1,132 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.EventGrid.Models;
+using Microsoft.Azure.Commands.EventGrid.Utilities;
+using Microsoft.Azure.Management.EventGrid.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+namespace Microsoft.Azure.Commands.EventGrid
+{
+    /// <summary>
+    /// 'Get-AzureRmEventGridDomainTopic' Cmdlet gives the details of a / List of EventGrid domain topic(s)
+    /// <para> If Domain topic name is provided, a single Domain topic details will be returned</para>
+    /// <para> If Domain topic name is not provided, list of all Domain topics under a domain will be returned</para>
+    /// </summary>
+    [Cmdlet(
+        VerbsCommon.Get,
+        ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "EventGridDomainTopic",
+        DefaultParameterSetName = DomainTopicNameParameterSet),
+    OutputType(typeof(PSDomainTopic), typeof(PSDomainTopicListInstance))]
+
+    public class GetAzureEventGridDomainTopic : AzureEventGridCmdletBase
+    {
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
```

Comment:
@ahamad-MS nit: same comment about `ValueFromPipelineByPropertyName`

Reply:
@cormacpayne Same here. This is still needed otherwise the following scenario will fail

Get-AzureRmEventGridDomain -ResourceGroupName aegBugbashResourceGroup2 -Name aegBugBashDomain2 | Get-AzureRmEventGridDomainTopic